File: 프로그래머스/1/155652. 둘만의 암호/둘만의 암호.java
import java.util.ArrayList;
class Solution {
    public String solution(String s, String skip, int index) {
                ArrayList<Integer> skiplist = new ArrayList<>();
        for (int i=0;i < skip.length();i++){
            for (char c: skip.toCharArray()) skiplist.add((int) c);
        }

        StringBuilder ans = new StringBuilder();
        for (char c: s.toCharArray()){
            int afteridx = (int)c + index;
            int endIdx = (int) 'z', startIdx = (int) 'a';
            
            for (int i = (int) c + 1; i <= afteridx; i++) {
                if (i > endIdx) {
                    i -= 26;
                    afteridx -= 26;
                }
                if (skiplist.contains(i)) afteridx++;
            }
            
            ans.append((char) afteridx);

        }

        return ans.toString();
    }
}

File: 프로그래머스/1/155652. 둘만의 암호/둘만의 암호.py
def solution(s, skip, index):
    skiplist = [ord(c) for c in skip]

    ans = []
    for c in s:
        afteridx = ord(c) + index
        endIdx = ord('z')
        startIdx = ord('a')

        i = ord(c) + 1
        while i <= afteridx:
            if i > endIdx:
                i -= 26
                afteridx -= 26
            
            if i in skiplist:
                afteridx += 1
            
            i += 1

        ans.append(chr(afteridx))

    return ''.join(ans)